### PR TITLE
Adds riot-v3.0.7

### LINKED ISSUE #85: Update riot to v3.0
http://riotjs.com/release-notes/

### PR DESCRIPTION
**Pull request by ckcollab**
Tested locally and seems to work! I'm going to run all results and make sure that works as well, will just take quite a while.

Resolves #85 and relates to https://github.com/riot/riot/issues/1978 and https://github.com/riot/riot/issues/2192

### CODE REVIEW

**Comment by PascalLeMerrer** [comment]
Thx for adding the benchmark for riot 3.
When trying to build and run it, I've got an error:

```
ERROR in ./src/app.html
    Module build failed: Error: Riot parser "js.babel" is not registered.
        at Object._req (/Users/pascallemerrer/Documents/Dev/projects/js-framework-benchmark/riot-v3.0.5/node_modules/riot-compiler/lib/parsers.js:110:13)
        at _compileJS (/Users/pascallemerrer/Documents/Dev/projects/js-framework-benchmark/riot-v3.0.5/node_modules/riot-compiler/lib/compiler.js:433:47)
        at getCode (/Users/pascallemerrer/Documents/Dev/projects/js-framework-benchmark/riot-v3.0.5/node_modules/riot-compiler/lib/compiler.js:780:10)
        at /Users/pascallemerrer/Documents/Dev/projects/js-framework-benchmark/riot-v3.0.5/node_modules/riot-compiler/lib/compiler.js:942:26
        at RegExp.[Symbol.replace] (native)
        at String.replace (native)
        at /Users/pascallemerrer/Documents/Dev/projects/js-framework-benchmark/riot-v3.0.5/node_modules/riot-compiler/lib/compiler.js:940:23
        at RegExp.[Symbol.replace] (native)
        at String.replace (native)
        at Object.compile (/Users/pascallemerrer/Documents/Dev/projects/js-framework-benchmark/riot-v3.0.5/node_modules/riot-compiler/lib/compiler.js:905:6)
        at Object.module.exports (/Users/pascallemerrer/Documents/Dev/projects/js-framework-benchmark/riot-v3.0.5/node_modules/tag-loader/index.js:6:51)
     @ ./src/main.es6.js 11:0-21
```

The steps I followed are:
```
cd riot-v3.0.5
npm install
npm run build-prod
```

File: riot-v3.0.5/src/app.html [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  }
+
+  this.rows = store.data
+  this.selected = store.selected
+
+  this.one('mount', () => {
+    // we need to replace the id of this DOM element
+    // riot automatically bind ids to the tags scopes but 'update' is a read only property
+    // that's a quick way to replace the id in order to fit the bench specs
+    this._update.setAttribute('id', 'update')
```
Comment (GianlucaGuarini) [inline riot-v3.0.5/src/app.html]: this is no longer needed

File: riot-v3.0.5/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+var loaders = [
+  {
+    test: /\.es6\.js$/,
+    loader: 'babel-loader',
+    exclude: /node_modules/
+  },
+  {
+    test: /\.html$/, loader: 'tag',
+    query: {
+      type: 'babel'
```
Comment (GianlucaGuarini) [inline riot-v3.0.5/webpack.config.js]: change `babel` to `es6`

File: riot-v3.0.5/src/app.html [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      this.rows = store.data
+      this.selected = store.selected
+      stopMeasure()
+    }
+  }
+
+  this.rows = store.data
+  this.selected = store.selected
+
+  this.one('mount', () => {
```
Comment (GianlucaGuarini) [inline riot-v3.0.5/src/app.html]: remove this listener

File: riot-v3.0.5/src/app.html (hunk 1 above)
Comment (ckcollab) [inline riot-v3.0.5/src/app.html]: fixed

File: riot-v3.0.5/webpack.config.js (hunk 2 above)
Comment (ckcollab) [inline riot-v3.0.5/webpack.config.js]: fixed

File: riot-v3.0.5/src/app.html (hunk 3 above)
Comment (ckcollab) [inline riot-v3.0.5/src/app.html]: removed

**Comment by ckcollab** [comment]
> Thx for adding the benchmark for riot 3.

@PascalLeMerrer no problem, my pleasure! The change from babel -> es6 seems to have fixed this, pushed the change.

File: riot-v3.0.5/src/app.html [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+              <button type='button' class='btn btn-primary btn-block' id='runlots' onclick={ methods.runLots }>Create 10,000 rows
+              </button>
+            </div>
+            <div class='col-sm-6 smallpad'>
+              <button type='button' class='btn btn-primary btn-block' id='add' onclick={ methods.add }>
+                Append 1,000 rows
+              </button>
+            </div>
+            <div class='col-sm-6 smallpad'>
+              <button type='button' class='btn btn-primary btn-block' id='_update' onclick={ methods.update }>Update every 10th row
```
Comment (GianlucaGuarini) [inline riot-v3.0.5/src/app.html]: change `_update` with `update`

Comment (ckcollab) [inline riot-v3.0.5/src/app.html]: _updated ;)

**Comment by PascalLeMerrer** [comment]
@ckcollab It's still broken, by another error:

```
TypeError: _riot2.default is undefined (main.js:75:2)
```

**Comment by ckcollab** [comment]
@PascalLeMerrer hmm, this is what I get running the command you mentioned previously:

image

**Comment by PascalLeMerrer** [comment]
Sorry, I was not clear. The build works indeed, this error occurs later, when I try to run the benchmark in a web browser.

Steps to reproduce:

- go to the project root
- npm start
- open a web browser on http://127.0.0.1/riot-v3.0.5

It should display the test runner, insted I've got a blank page and the error message is the console.

By the way, the page at http://127.0.0.1 misses a link to the test of Riot 3.0.5

**Comment by GianlucaGuarini** [comment]
@PascalLeMerrer @ckcollab hold on merging this. I will make a new patch release to fix the interop export issues soon

**Comment by ckcollab** [comment]
@PascalLeMerrer added link from main page to riot 3.0.5, thanks for pointing that out.

Can't wait to see what @GianlucaGuarini is cookin up 👍

**Comment by GianlucaGuarini** [comment]
@PascalLeMerrer you can now update and merge it should be fine

**Comment by PascalLeMerrer** [comment]
@ckcollab did you run the tests?

**Comment by ckcollab** [comment]
@PascalLeMerrer Yeah I get errors now, may have changed something unintentionally but does not seem to be working any more:

```bash
$ npm run selenium -- --framework riot-v3

> webdriver-ts@1.0.0 selenium /Users/eric/src/js-framework-benchmark/webdriver-ts
> node dist/benchmarkRunner.js "--framework" "riot-v3"

{ _: 
   [ '/usr/local/bin/node',
     '/Users/eric/src/js-framework-benchmark/webdriver-ts/dist/benchmarkRunner.js' ],
  help: false,
  framework: [ 'riot-v3' ],
  check: 'false',
  '$0': 'dist/benchmarkRunner.js' }
target directory results
Frameworks that will be benchmarked [ { name: 'riot-v3.0.7', uri: 'riot-v3.0.7', useShadowRoot: false } ]
Benchmarks that will be run [ '01_run1k',
  '02_replace1k',
  '03_update10th1k',
  '04_select1k',
  '05_swap1k',
  '06_remove-one-1k',
  '07_create10k',
  '08_create1k-after10k',
  '09_clear10k',
  '10_clear-2nd-time10k',
  '21_ready-memory',
  '22_run-memory' ]
benchmarking  { name: 'riot-v3.0.7', uri: 'riot-v3.0.7', useShadowRoot: false } 01_run1k
(node:9484) DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
after initialized  01_run1k 0 riot-v3.0.7
retry failed
retry failed
retry failed
retry failed
retry failed
retry failed
QUIT

/Users/eric/src/js-framework-benchmark/webdriver-ts/node_modules/selenium-webdriver/lib/promise.js:654
    throw error;
    ^
Retrying failed

```

**Comment by PascalLeMerrer** [comment]
I have the exact same issue, whatever the test I run is (not limited to riot-v3)

**Comment by krausest** [comment]
Thanks for your work. Runs fine on my machine. Riot 3.0.7 performs much better than 2.6.7 in this benchmark:
bildschirmfoto 2017-01-13 um 18 57 20
@PascalLeMerrer @ckcollab 
Did you run "npm start" in the project root and start selenium in a different shell (the http-server must be running)? Can you open riot in the browser (http://localhost:8080/riot-v3.0.7/)?

**Comment by PascalLeMerrer** [comment]
@krausest: yes I did.
And I also set the LANG environment variable as I'm OSX in a non english version

**Pull request merged**